src/azul/service/manifest_service.py

```diff
@@ -514,7 +514,10 @@ def command_lines(self,
                       manifest: Optional[Manifest],
                       url: str
                       ) -> Optional[JSON]:
-        generator_cls = ManifestGenerator.cls_for_format[manifest.format_]
+        # command_lines returns callback urls to the service when a manifest
+        # isn't provided, so format_ can be set to a default like compact.
+        format_ = ManifestFormat.compact if manifest is None else manifest.format_
```

Comment:
This is confusing and could break if the compact generator overrode `command_lines`. If you don't know the format, you should set `generator_cls` to `ManifestGenerator`.